### DISCUSSION TITLE
Replication setup

### DISCUSSION THREAD

**Discussion by xantinium** [apache/kvrocks#2674] [Q&A] [answered]
Hi there!
I'm trying to setup 2 instances of KVRocks (master and replica) to test failover behaviour.
The master instance is running on **port=6666** and the replica is running on **port=6667** with configured **slaveof** param. Everything fine, replica is connected successfully:
`New replica: 127.0.0.1:49886 was added, start incremental syncing`

I expect that, after stopping the master instance, the replica will be promoted and replace disconnected master. But I see it only tries to reconnect:
`[replication] Failed to connect the master, err: Operation now in progress`

Is there a way to achieve the behaviour of automatic promotion?

**Comment by PragmaTwice** [comment, marked as answer]
You can use the sentinel mechanism: https://redis.io/docs/latest/operate/oss_and_stack/management/sentinel/

Also Kvrocks Controller provides such capability (if you have a large cluster): https://github.com/apache/kvrocks-controller. (if you just have one master and one or several replicas, I recommend the sentinel way.)

**Reply by xantinium** [reply]
Thanks a lot!
